Fix infinite looping in skipArray/skipObject

leadpony opened the pull request:
JSON array and objcet which are not properly closed cause infinite looping 
in JsonParser#skipArray() and JsonParser#skipObject() 

Signed-off-by: leadpony <dev@leadpony.org>

leadpony commented:
@m0mus 
Thank you for your review.
When I wrote and committed the modification, it was on 30 Dec 2018. Therefore I believe the copyright year should be 2018 as is, not 2019.

lukasj commented on impl/src/main/java/org/glassfish/json/JsonParserImpl.java:
```diff
@@ -501,6 +501,10 @@ void skip() {
                         depth++;
                         break;
+                    case EOF:
+                        throw parsingException(JsonToken.EOF, "[CURLYCLOSE]");
```
this is not exactly true per JSON grammar since after  'CURLYOPEN' there can be also 'STRING'. Generally speaking, the error should list all valid options.

lukasj commented on impl/src/main/java/org/glassfish/json/JsonParserImpl.java:
```diff
@@ -561,6 +565,10 @@ void skip() {
                         depth++;
                         break;
+                    case EOF:
+                        throw parsingException(JsonToken.EOF, "[SQUARECLOSE]");
```
same here

leadpony commented:
Hello @lukasj 
Thank you for your kind help.
The original  `skipArray()` and `skipObjec()` did not throw any exceptions even when the input JSON is not well-formed. I changed the behavior and make them detect all such errors. I also changed Javadoc comment for `JsonParser` API  interface and copyright years of files I updated today.

leadpony commented:
I have modified `skipObject()` to emit more appropriate message.

leadpony commented:
I abandon this PR.
